integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/api/RestApiAdminServiceTestCase.java

```diff
@@ -101,30 +124,109 @@ public void testGetAPIForTenant() throws Exception {
     }
 
     @Test(groups = { "wso2.esb" },
-          description = "Test API update service",
+          description = "Test API update service with string param",
           priority = 5)
-    public void testUpdateAPI() throws Exception {
+    public void testUpdateAPIFromString() throws Exception {
         String updateData =
                 "<api xmlns=\"http://ws.apache.org/ns/synapse\" name=\"SampleServiceAPI\" context=\"/createApi\">\n"
                         + "    <resource methods=\"POST PUT\" uri-template=\"/\"> </resource></api>";
         boolean apiUpdated = restAdminClient.updateAPIFromString(api1.getName(), updateData);
+        APIData updatedAPI = restAdminClient.getAPIbyName(api1.getName());
+        ResourceData[] updatedResourceData = updatedAPI.getResources();
+        String updatedResource = restAdminClient.getAPIResource(updatedResourceData[0]);
         Assert.assertEquals(apiUpdated, true, "API was not updated");
+        Assert.assertTrue(updatedResource.contains("POST PUT"), "API was not updated with new method");
     }
 
     @Test(groups = { "wso2.esb" },
-          description = "Test API update service for tenant",
+          description = "Test API update service with API data param",
           priority = 6)
+    public void testUpdateAPIFromAPIData() throws Exception {
+        boolean apiUpdated = restAdminClient.updateAPIFromAPIData(api1.getName(), api3);
+        String[] sequences = restAdminClient.getAPISequences();
+        Assert.assertEquals(apiUpdated, true, "API was not updated");
+        Assert.assertNotNull(sequences, "API does not contained updated sequence");
```

Comment:
Shall we change this to "API does not contain updated sequence" ?

Reply:
Fixed with 8ea24440b5d2efb4a8940c1bd861c1a2f809c00f